junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/TestMethodTestDescriptor.java

```diff
@@ -230,4 +234,48 @@ private void invokeAfterEachCallbacks(JupiterEngineExecutionContext context) {
 		});
 	}
 
+	@Override
+	public void nodeSkipped(JupiterEngineExecutionContext context, TestDescriptor descriptor, SkipResult result) {
+		invokeTestWatchers(context, t -> {
+			t.testSkipped(createTestIdentifier(), result.getReason(), context.getExtensionContext());
+		});
+	}
+
+	@Override
+	public void nodeFinished(JupiterEngineExecutionContext context, TestDescriptor descriptor,
+			TestExecutionResult result) {
+		invokeTestWatchers(context, t -> {
+			ExtensionContext extensionContext = context.getExtensionContext();
+			Status status = result.getStatus();
+			String identifier = createTestIdentifier();
+			switch (status) {
+				case SUCCESSFUL:
+					t.testSuccessful(identifier, extensionContext);
+					break;
+				case FAILED:
+					t.testFailed(identifier, result.getThrowable().get(), extensionContext);
+					break;
+				case ABORTED:
+					t.testAborted(identifier, result.getThrowable().get(), extensionContext);
+					break;
+			}
+		});
+	}
+
+	private void invokeTestWatchers(JupiterEngineExecutionContext context, Consumer<TestWatcher> callback) {
+		ExtensionRegistry registry = context.getExtensionRegistry();
+		registry.getReversedExtensions(TestWatcher.class).forEach(t -> {
+			try {
+				callback.accept(t);
+			}
+			catch (Exception e) {
+				//swallow
+			}
+		});
+	}
+
+	private String createTestIdentifier() {
+		Method testMethod = this.getTestMethod();
+		return String.format("%s#%s", testMethod.getDeclaringClass().getName(), testMethod.getName());
```

Comment:
This is inappropriate for a few reasons.

1. The declaring class may not be the current test class, which would lead to conflicts for IDs for the same method executed multiple times in a class hierarchy.
2. The method parameters are missing, which would lead to conflicts for overloaded test methods.

If we were to do this, we'd use `ReflectionUtils.getFullyQualifiedMethodName(Class<?>, String, Class<?>...)`, but as @marcphilipp already suggested, we should actually remove this "ID" since the `ExtensionContext` provides sufficient contextual information.